### PR TITLE
Add ArangoDB Container support

### PR DESCRIPTION
**Pull request by nshine**
ArangoDB -> https://www.arangodb.com

### CODE REVIEW

**Comment by thedrow** [comment]
Thanks!

Celery definitely needs this one for its integration suite.

**Comment by tillahoffmann** [comment]
The failing tests seem to be the consequence of a recent sphinx change (https://github.com/sphinx-doc/sphinx/issues/10474). You should be able to fix it by changing `language = None` to `language = 'en'` in `docs/conf.py`.

**Comment by nshine** [comment]
Fixed

File: setup.py [hunk 1]
```diff
@@ -67,6 +67,7 @@
         'rabbitmq': ['pika'],
         'clickhouse': ['clickhouse-driver'],
         'keycloak': ['python-keycloak'],
+        'arangodb': ['python-arango']
```
Comment (tillahoffmann) [inline setup.py:70]:
```suggestion
        'arangodb': ['python-arango'],
```
Nit: this just makes the line changes a bit easier to read because the comma is already there when the next container is added.

File: testcontainers/arangodb.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+class ArangoDbContainer(DbContainer):
+    """
+    ArangoDB container.
+
+    Example
+    -------
+    The example will spin up a ArangoDB container.
+    You may use the :code:`get_connection_url()` method which returns a arangoclient-compatible url
+    in format :code:`scheme://host:port`. As of now, only a single host supported (over HTTP).
```
Comment (tillahoffmann) [inline testcontainers/arangodb.py:17]:
```suggestion
    in format :code:`scheme://host:port`. As of now, only a single host is supported (over HTTP).
```

File: testcontainers/arangodb.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        with ArangoContainer("arangodb:3.9.1") as arango:
+            client = ArangoClient(hosts=arango.get_connection_url())
+
+            # Connect
+            sys_db = arango_client.db(username='root', password='')
+
+            # Create a new database named "test".
+            sys_db.create_database("test")
+    """
+    def __init__(self, image="arangodb:latest", **kwargs):
```
Comment (tillahoffmann) [inline testcontainers/arangodb.py:29]:
```suggestion
    def __init__(self, image="arangodb:latest", port_to_expose=8529, **kwargs):
```

File: testcontainers/arangodb.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        super().__init__(image=image)
+        self.port_to_expose = 8529
```
Comment (tillahoffmann) [inline testcontainers/arangodb.py:31]:
```suggestion
        self.port_to_expose = port_to_expose
```

File: testcontainers/arangodb.py (hunk 3 above)
Comment (tillahoffmann) [reply testcontainers/arangodb.py:29]: This makes it easier to modify the exposed port if necessary (we should also change that on a bunch of other containers at some point).

File: testcontainers/arangodb.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        self.with_exposed_ports(self.port_to_expose)
+
+        self.arango_host = kwargs.get('ARANGO_HOST', 'localhost')
+
+        # https://www.arangodb.com/docs/stable/deployment-single-instance-manual-start.html
+        self.arango_no_auth = kwargs.get(
```
Comment (tillahoffmann) [inline testcontainers/arangodb.py:37]: Shall we also move these to proper keyword arguments? That should make it a bit easier for code completion, for example. Something like this?

```python
# arango_no_auth=None in the __init__ signature.
self.arango_no_auth = arango_no_auth or os.environ.get("ARANGO_NO_AUTH")
```

File: testcontainers/arangodb.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        port = self.get_exposed_port(self.port_to_expose)
+        url = f"{scheme}://{self.arango_host}:{port}"
+
+        return url
+
+    def _connect(self):
+        wait_for_logs(
+            self,
+            predicate="is ready for business",
+            timeout=10)
```
Comment (tillahoffmann) [inline testcontainers/arangodb.py:64]: Let's use `TC_MAX_TRIES` as the timeout. It's somewhat obscurely hidden, sorry.

https://github.com/testcontainers/testcontainers-python/blob/f9b7374e637b042d97be4891b54a496e0f9f0b96/testcontainers/core/config.py#L3

**Review by tillahoffmann — COMMENTED** [review]
Looks great! A few small comments.

**Comment by nshine** [comment]
Thanks for your review.  updated.

File: testcontainers/arangodb.py (hunk 5 above)
Comment (tillahoffmann) [reply testcontainers/arangodb.py]: Makes sense. 👍 We should probably update that in mysql at some point.

File: testcontainers/arangodb.py [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+            # Create a new database named "test".
+            sys_db.create_database("test")
+    """
+    def __init__(self,
+                 image="arangodb:latest",
+                 port_to_expose=8529,
+                 hostname='localhost',
+                 arango_root_password='passwd',
+                 arango_no_auth=None,
```
Comment (tillahoffmann) [inline testcontainers/arangodb.py:35]: For `arango_no_auth` and `arango_random_root_password`, would it be possible to turn them into booleans (defaulting to `False`) to make the interface "more pythonic" (I appreciate it's not a very well defined concept)? Then the user doesn't have to know that passing a string of `1` will have the desired effect, but they can rely on the python semantics of `True` or `False`. What do you think?

File: testcontainers/arangodb.py [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+from os import environ
+from testcontainers.core.config import MAX_TRIES
+from testcontainers.core.generic import DbContainer
+from testcontainers.core.waiting_utils import wait_for_logs
+
+
+class ArangoDbContainer(DbContainer):
+    """
+    ArangoDB container.
+
```
Comment (tillahoffmann) [inline testcontainers/arangodb.py:13]: If we could document the parameters of the constructor here, that'd be great!

**Review by tillahoffmann — COMMENTED** [review]
Few more thoughts.

File: testcontainers/arangodb.py (hunk 7 above)
Comment (nshine) [reply testcontainers/arangodb.py]: I think that by definition of falsy values also includes None (and empty collections, etc.) - I'm fine with aligning to your request - However I think even what we did here (None) is tricky concept and should be default to actual string of "0". 
for the following reasons:
1. aligning to ArangoDB documentation 
    https://www.arangodb.com/docs/stable/deployment-single-instance-manual-start.html

2. since it env var. we can't do os.environ['KEY'] = False/None (is it do expect string values) - I do understand that it not the same as it eventually passing to Docker through -e , but we do need to remember it originated in env variable . because we wrap the Docker python-sdk, it might be OK (they take care the convert of the Falsy env params)

TLDR;
- Resolved. however, I do have some concerns. (above)
- Removed additional input (hostname) -> not relevant as of now and can be "calculated" from matched internal func

**Comment by nshine** [comment]
Hi, all resolved.

**Comment by nshine** [comment]
@tillahoffmann - I will appreciate if you could merge and generate a new release.

**Comment by nshine** [comment]
Hi @thedrow / @SergeyPirogov  any chance you can step in and approve&release this one ?

**Review by SergeyPirogov — APPROVED** [review]

**Comment by SergeyPirogov** [comment]
@nshine thank you

**Comment by nshine** [comment]
Hi @SergeyPirogov 
Thanks for the approve - can you please trigger a new release? i.e. 3.6.0
